SYCL/Matrix/element_wise_int8_sumLocalCol.cpp

```diff
@@ -0,0 +1,237 @@
+// Kernel B sum by col
+#include <iostream>
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental::matrix;
+
+#define SG_SZ 16
+
+#define TN SG_SZ
+#define TK 32
+
+template <typename T, size_t NUM_ROWS, size_t NUM_COLS> struct big_matrix {
+public:
+  T *mat;
+
+public:
+  T *get_data() { return mat; }
+  void set_data(T *data) { mat = data; }
+  big_matrix(T *data) : mat(data) {}
+};
+
+template <typename T, size_t M, size_t N>
+void sum_cols_ref(
+    accessor<T, 2, access::mode::read, access::target::host_buffer> B,
+    accessor<int, 1, access::mode::read, access::target::host_buffer>
+        sum_cols) {
+  int sum_cols_ref[N] = {0};
+  for (size_t j = 0; j < N; j++) {
+    for (size_t i = 0; i < M; i++) {
+      sum_cols_ref[j] += B[i][j];
+    }
+    auto diff = sum_cols[j] - sum_cols_ref[j];
+    assert(std::fabs(static_cast<int>(diff)) <=
+           std::numeric_limits<int>::epsilon());
+  }
+}
+
+template <typename T, size_t M, size_t N>
+void matrix_sum_cols(queue q, big_matrix<T, M, N> &B, nd_range<2> &r) {
+  buffer<int8_t, 2> bufB(B.get_data(), range<2>(M, N));
+  // size of vector is known because SG size of set by the user in this case
+  int sum_cols[N] = {0};
+  buffer<int> sum_cols_v(sum_cols, N); // there are total of tK/4 * 2, 16 rows
+  q.submit([&](handler &cgh) {
+     auto accB = bufB.get_access<access::mode::read_write>(cgh);
+
+     auto v = sum_cols_v.get_access<access::mode::atomic>(cgh);
```

Comment:
nit: use accessor's CTAD rules:

```suggestion
     accessor accB{bufB, cgh};
     accessor v{sum_cols_v, cgh};
```

Reply:
@aelovikov-intel How can I have an accessor of an `atomic` type?  My accessor `v` has to be atomic as I call an `atomic_fetch_add()` on it later.

Reply:
If I'm reading https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_accessor_specialization_with_access_modeatomic correctly, these are deprecated, and we shouldn't be adding them in new tests.

I believe https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:atomic-references is the interface that should be used instead.